### ISSUE TITLE
Moving token from body to header

### ISSUE THREAD

**Issue by IljaN** [lexik/LexikJWTAuthenticationBundle#313]
I would like to have the token which is returned after login, to be returned in the header instead of the body.

I created a AuthenticationSuccessListener, but can`t overide the body to make it empty. Is this intentional?
```php
    public function onAuthenticationSuccessResponse(AuthenticationSuccessEvent $event) {
        $data = $event->getData();

        if (!$event->getUser() instanceof UserInterface) {
            return;
        }

        $response = $event->getResponse();
        $response->headers->add(['authorization' => $data['token']]); // works
        $response->setContent(''); // no effect
        $event->setData([]); // no effect
    }
```

**Comment by chalasr** [comment]
This is the expected behavior indeed, but that's because we didnot see any valid use case at the moment it has been implemented. I'll have a look at what we can do, maybe this event deserves a `setResponse()` as well.

**Comment by IljaN** [comment]
My usecase is that for example vue-auth defaults to reading the token from the header after login. I googled and found some other jwt implementations that also behave like that altough they seem to be in the minority.

**Comment by chalasr** [comment]
Fixed by #320

**Comment by IljaN** [comment]
@Batch1211 Thank you very much! <3
